vespajlib/src/main/java/com/yahoo/tensor/TensorType.java

```diff
@@ -69,8 +72,10 @@ public static Value fromId(String valueTypeString) {
             switch (valueTypeString) {
                 case "double" : return Value.DOUBLE;
                 case "float" : return Value.FLOAT;
-                default : throw new IllegalArgumentException("Value type must be either 'double' or 'float'" +
-                                                             " but was '" + valueTypeString + "'");
+                case "bfloat16" : return Value.BFLOAT16;
```

Comment:
consider looping over Value.values() and comparing with their "id" instead of repeating the names here.

Reply:
Thanks, fixed.